SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/OptOutServices/EmailServiceV1.swift

```diff
@@ -0,0 +1,209 @@
+
+    public init(email: String, attemptId: String) {
+        self.email = email
+        self.attemptId = attemptId
+    }
+}
+
+public struct EmailDataResponseV1: Codable {
+    public let items: [EmailDataResponseItemV1]
+}
+
+public struct EmailDataResponseItemV1: Codable {
+    public let email: String
+    public let attemptId: String
+    public let status: EmailStatusV1
+    public let errorCode: EmailErrorCodeV1?
+    public let data: [EmailDatumV1]
+    public let emailReceivedAt: TimeInterval?
+
+    enum CodingKeys: String, CodingKey {
+        case email, attemptId, status, data
+        case errorCode = "error_code"
+        case emailReceivedAt = "email_received_at"
+    }
+
+    public var confirmationLink: String? {
+        data.first(where: { $0.name == "link" })?.value
+    }
+
+    public var linkObtainedOnBEDate: Date? {
+        guard let emailReceivedAt else { return nil }
+        return Date(timeIntervalSince1970: emailReceivedAt)
+    }
+}
+
+public enum EmailStatusV1: String, Codable {
+    case ready
+    case pending
+    case unknown
+    case error
+}
+
+public enum EmailErrorCodeV1: String, Codable {
+    case serverError = "server_error"
+    case extractionError = "extraction_error"
+    case requestError = "request_error"
+}
+
+public struct EmailDatumV1: Codable {
+    public let name: String
+    public let value: String
+}
+
+public struct DeleteEmailDataRequestV1: Codable {
+    public let items: [EmailDataRequestItemV1]
+
+    public init(items: [EmailDataRequestItemV1]) {
+        self.items = items
+    }
+}
+
+public protocol EmailServiceV1Protocol {
+    func fetchEmailData(items: [EmailDataRequestItemV1]) async throws -> EmailDataResponseV1
+    func deleteEmailData(items: [EmailDataRequestItemV1]) async throws
+}
+
+public struct EmailServiceV1: EmailServiceV1Protocol {
+    public struct Constants {
+        static let endpointSubPath = "/dbp/em/v1"
+        static let maxBatchSize = 100
+    }
+
+    public let urlSession: URLSession
+    private let authenticationManager: DataBrokerProtectionAuthenticationManaging
+    private let settings: DataBrokerProtectionSettings
+    private let servicePixel: DataBrokerProtectionBackendServicePixels
+
+    public init(urlSession: URLSession = URLSession.shared,
+                authenticationManager: DataBrokerProtectionAuthenticationManaging,
+                settings: DataBrokerProtectionSettings,
+                servicePixel: DataBrokerProtectionBackendServicePixels) {
+        self.urlSession = urlSession
+        self.authenticationManager = authenticationManager
+        self.settings = settings
+        self.servicePixel = servicePixel
+    }
+
+    public func fetchEmailData(items: [EmailDataRequestItemV1]) async throws -> EmailDataResponseV1 {
+        guard !items.isEmpty else {
+            throw EmailErrorV1.noEmailData
+        }
+
+        guard items.count <= Constants.maxBatchSize else {
+            throw EmailErrorV1.batchSizeExceeded
```

Comment:
I'd be tempted to put an assertionFailure in here since this should never happen, it might help us catch it if it ever does whilst developing